crypto/merkle/tree.go

```diff
@@ -59,7 +66,7 @@ func HashFromByteSlices(items [][]byte) []byte {
 // Finally, considering that the recursive implementation is easier to
 // read, it might not be worthwhile to switch to a less intuitive
 // implementation for so little benefit.
-func HashFromByteSlicesIterative(input [][]byte) []byte {
+func hashFromByteSlicesIterative(input [][]byte) []byte {
```

Comment:
[blocking]

This is technically Go API breaking so the conventional commit for this PR should include `!` as in: "perf(crypto/merkle)!: reduce allocations..."

On a related note, we still don't have a strategy on how to cut breaking releases of celestia-core so I think un-exporting this will delay it's inclusion of getting this PR into a release. IMO it would be easier to keep it exported and add a deprecation notice like:

```go
// Finally, considering that the recursive implementation is easier to
// read, it might not be worthwhile to switch to a less intuitive
// implementation for so little benefit.
//
// Deprecated: use HashFromByteSlices instead.
func HashFromByteSlicesIterative(input [][]byte) []byte {
```

that way we can merge the PR now and consider un-exporting it when we figure out how to do breaking changes of this repo.

Reply:
Added (+ fixed doc comment).